src/compiler/utilities.ts

```diff
@@ -101,7 +101,12 @@ namespace ts {
     }
 
     export function changesAffectModuleResolution(oldOptions: CompilerOptions, newOptions: CompilerOptions): boolean {
-        return oldOptions.configFilePath !== newOptions.configFilePath || moduleResolutionOptionDeclarations.some(o =>
+        return oldOptions.configFilePath !== newOptions.configFilePath ||
+            optionsHaveModuleResolutionChanges(oldOptions, newOptions);
+    }
+
+    export function optionsHaveModuleResolutionChanges(oldOptions: CompilerOptions, newOptions: CompilerOptions) {

```

Comment:
@sheetalkamat would it be reasonable to expose this function in the public API? That would allow API users to reuse their ModuleResolutionCache until changes in CompilerOptions require a new cache.

Reply:
Sure. Note that this only compares CompilerOptions and not changes in directory structure which could affect the module resolution.